clarity-ext-scripts/clarity_ext_scripts/covid/partner_api_client.py

```diff
@@ -0,0 +1,144 @@
+
+from datetime import datetime
+from luhn import verify as mod10verify
+from luhn import generate as mod10generate
+import logging
+import requests
+from requests import ConnectionError, Timeout
+from retry import retry
+import re
+
+
+log = logging.getLogger(__name__)
+
+COVID_RESPONSE_POSITIVE = "positive"
+COVID_RESPONSE_NEGATIVE = "negative"
+COVID_RESPONSE_FAILED = "failed"
+
+VALID_COVID_RESPONSES = {COVID_RESPONSE_POSITIVE,
+                         COVID_RESPONSE_NEGATIVE, COVID_RESPONSE_FAILED}
+
+
+class PartnerAPISampleInformation(object):
+
+    def __init__(self, referral_code, lab_referral, arrival_date, result_date, comment, cov19_result):
+        try:
+            if not isinstance(referral_code, str) and not len(referral_code) == 10:
+                raise AssertionError(
+                    "referral_code needs to be 10 digit number string.")
+
+            if not mod10verify(referral_code):
+                raise AssertionError("Check code digit {} (last digit) did not match mod10 requirement"
+                                     "in referral code: {}. Expected: {}".format(referral_code[-1],
+                                                                                 referral_code,
+                                                                                 mod10generate(referral_code[:-1])))
```

Comment:
I think that the `Expected` part could give misleading information. As I understand it, this is only helpful if the code is correct but the error code part is wrong.

So let's say the user is supposed to send in "111111111x" but sends in "211111111y". The real error is the 1 that became 2, but the user get's an error telling them to send in the string "2111111111z".

Reply:
You are right. I'll fix that.